assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -5075,4 +5040,460 @@ public String intToTime(int time) {
     public String getContentReviewServiceName() {
         return this.contentReviewService.getServiceName();
     }
+
+    @Override
+    @Transactional
+    public void gradeSubmission(AssignmentSubmission submission, String gradeOption, Map<String, Object> options, List<String> alerts) {
```

Comment:
instead of passing an AssignmentSubmission object on a public api probably better to pass in a submission Id